index.js

```diff
@@ -267,14 +267,18 @@ let runInFiberContext = function (testInterfaceFnName, before, after, fnName) {
     let runHook = function (hookFn) {
         return origFn(function (done) {
             executeHooksWithArgs(before)
-                .then(() => new Promise((resolve) => {
+                .then(() => new Promise((resolve, reject) => {
                     return Fiber(() => {
-                        hookFn.call(this)
+                        try {
+                            hookFn.call(this)
+                        } catch (e) {
+                            reject(e)
+                        }
                         resolve()
                     }).run()
                 }))
                 .then(() => executeHooksWithArgs(after))
-                .then(() => done(), () => done())
+                .then(done, done)
```

Comment:
@georgecrawford I don't think this would be a good idea as it would let the test case fail when there is an error in the afterTest hook. Right now it is designed in a way that errors in hooks don't affect the result of the test at all.